ytypes/node.go

```diff
@@ -59,6 +59,10 @@ type retrieveNodeArgs struct {
 	// values are skipped (non-existing GoStructs are still initialized),
 	// and any value retrieved is always nil.
 	shadowPath bool
+	// reverseShadowPath reverses the meaning of the "path" and
+	// "shadow-path" tags when both are present while processing a
+	// GoStruct.
+	reverseShadowPath bool
```

Comment:
`preferShadowPath` perhaps given that we're not really "reversing" a path (this would say to me that `/a/b/c` becomes `/c/b/a`.

Reply:
The meaning of "reverse" here is to indicate that the path tags of "path" and "shadow-path" are effectively swapped during processing. The reason why I didn't use `preferShadowPath` is that it sounds somewhat unclear to me. What this flag does is that during unmarshalling, "shadow-path" is now unmarshalled, and "path" is silently ignored as a no-op. While `preferShadowPath` is a fine description for this behaviour, I think `reverseShadowPath` is very clear on what the behaviour is.

Reply:
SGTM, thanks for the clarification here.